protocols/luci-proto-sstp/htdocs/luci-static/resources/protocol/sstp.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+'require uci';
+'require form';
+'require network';
+
+network.registerPatternVirtual(/^sstp-.+$/);
+
+return network.registerProtocol('sstp', {
+	getI18n: function() {
+		return _('SSTP');
+	},
+
+	getIfname: function() {
+		return this._ubus('l3_device') || 'sstp-%s'.format(this.sid);
+	},
+
+	getOpkgPackage: function() {
+		return 'sstp-client';
+	},
+
+	isFloating: function() {
+		return true;
+	},
+
+	isVirtual: function() {
+		return true;
+	},
+
+	getDevices: function() {
+		return null;
+	},
+
+	containsDevice: function(ifname) {
+		return (network.getIfnameOf(ifname) == this.getIfname());
+	},
+
+	renderFormOptions: function(s) {
+		var dev = this.getL3Device() || this.getDevice(), o;
+
+		// -- general ---------------------------------------------------------------------
+
+		o = s.taboption('general', form.Value, 'server', _('SSTP Server'));
+		o.datatype = 'host';
+
+		o = s.taboption('general', form.Value, 'username', _('PAP/CHAP username'));
+
+		o = s.taboption('general', form.Value, 'password', _('PAP/CHAP password'));
+		o.password = true;
+
+		// -- advanced --------------------------------------------------------------------
+
+		o = s.taboption('advanced', form.Flag, 'ipv6', _('IPv6 support'), _('If checked, adds "+ipv6" to the pppd options'));
+
+		o = s.taboption('advanced', form.Value, 'log_level', _('sstpc log-level'));
```

Comment:
could we make a dropdown with loglevels that are available? So I do not have to remember which number is which loglevel?

Reply:
Added dropdown. Using numerical log level, as there is no specific description of each log-level as per documentation: https://www.mankier.com/8/sstpc#--log-level